formant_ros2_adapter/scripts/main.py

```diff
@@ -1189,202 +1193,299 @@ def handle_formant_teleop_msg(self, msg):
+            slot_type = service_request_slots[0]
+
+            # To do:
+            # Using numpy types seems like a good idea, but I can't get it working
+            # Error:
+            # AssertionError: The 'input' field must be of type 'int'
+            # if slot_type == "float32":
+            #     service_request_value = np.float32(command_text)
+            # elif slot_type == "float64":
+            #     service_request_value = np.float64(command_text)
+            # elif slot_type == "int8":
+            #     service_request_value = np.int8(command_text)
+            # elif slot_type == "int16":
+            #     service_request_value = np.int16(command_text)
+            # elif slot_type == "int32":
+            #     service_request_value = np.int32(command_text)
+            # elif slot_type == "int64":
+            #     service_request_value = np.int64(command_text)
+            # elif slot_type == "uint8":
+            #     service_request_value = np.uint8(command_text)
+            # elif slot_type == "uint16":
+            #     service_request_value = np.uint16(command_text)
+            # elif slot_type == "uint32":
+            #     service_request_value = np.uint32(command_text)
+            # elif slot_type == "uint64":
+            #     service_request_value = np.uint64(command_text)
+
+            if "int" in slot_type:
+                # If user sends a float, convert to int
+                service_request_value = int(float(command_text))
+            elif "float" in slot_type:
+                service_request_value = float(command_text)
+            else:
+                service_result = (
+                    "WARNING: Unsupported parameter type for numeric service " +
+                    f"{service_command}: " +
+                    f"{slot_type}"
+                )
+                print(service_result)
+                return success, service_result
+            # Set the attribute on the request to the command text
+            setattr(
+                service_request,
+                service_request_attribute,
+                service_request_value,
+            )
+
         else:
-            print("WARNING: No ROS2 publisher found for stream " + stream_name)
+            service_result = (
+                "WARNING: Unsupported ROS 2 service parameters for command " +
+                f"{service_command}"
+            )
+            print(service_result)
+            return success, service_result
+
+        # Check if the service is available
+        if service_client.wait_for_service(SERVICE_CALL_TIMEOUT) == False:
+            service_result = ("WARNING: Timeout waiting for service")
+            print(service_result)
+            return success, service_result
+        # Call the service if the paramaters are valid
+        if service_result == "":
+            service_result = service_client.call(service_request)
```

Comment:
service_client.call or service_client.call_async?

Reply:
That was on purpose: we want it to wait for and call the service in a simpler way